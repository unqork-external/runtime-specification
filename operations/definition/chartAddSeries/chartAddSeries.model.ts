import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { ChartAddSeriesOperationOptions } from './chartAddSeries.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@Stable()
@OpsBuilderStable()
@Description('Adds new series to a Chart component.')
@DiscriminatorValue(OperationTypes.CHART_ADD_SERIES)
export class ChartAddSeriesOperation extends Operation<OperationTypes.CHART_ADD_SERIES> {
  @Required()
  public type = OperationTypes.CHART_ADD_SERIES as const

  @Required()
  public options: ChartAddSeriesOperationOptions = new ChartAddSeriesOperationOptions()
}
