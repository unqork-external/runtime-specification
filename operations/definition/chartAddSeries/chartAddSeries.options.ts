import { CollectionOf, Description, Example, Required } from '@tsed/schema'

import { ChartSeries } from '../../../components'

export class ChartAddSeriesOperationOptions {
  @Required()
  @Description('The Property ID for the target Chart component.')
  targetKey: string

  @Description(
    'The index to insert the newly created series, starting at 0. If unset, adds the series to the end. Existing series on the same index will be pushed below. Negative indexes add from the bottom.',
  )
  @Example(0)
  targetIndex?: number

  @Required()
  @Description('The new series to add.')
  @CollectionOf(ChartSeries)
  series: ChartSeries[]
}
