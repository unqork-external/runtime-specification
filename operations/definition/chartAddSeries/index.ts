export * from './chartAddSeries.model'
export * from './chartAddSeries.options'
