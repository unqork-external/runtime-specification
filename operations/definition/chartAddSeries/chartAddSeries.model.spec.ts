import { ChartAddSeriesOperation } from './chartAddSeries.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ChartAddSeries Tests', () => {
  it('ChartAddSeries Model', () => {
    const { schema } = generateSchemaAndValidate(ChartAddSeriesOperation)

    expect(schema).toMatchSnapshot()
  })
})
