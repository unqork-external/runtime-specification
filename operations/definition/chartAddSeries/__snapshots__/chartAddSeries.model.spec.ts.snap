// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: ChartAddSeries Tests ChartAddSeries Model 1`] = `
{
  "definitions": {
    "ChartAddSeriesOperationOptions": {
      "properties": {
        "series": {
          "description": "The new series to add.",
          "items": {
            "$ref": "#/definitions/ChartSeries",
          },
          "type": "array",
        },
        "targetIndex": {
          "description": "The index to insert the newly created series, starting at 0. If unset, adds the series to the end. Existing series on the same index will be pushed below. Negative indexes add from the bottom.",
          "examples": [
            0,
          ],
          "type": "number",
        },
        "targetKey": {
          "description": "The Property ID for the target Chart component.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "series",
      ],
      "type": "object",
    },
    "ChartPropertyMap": {
      "properties": {
        "x": {
          "description": "The property name of the X values for the data of the series.",
          "minLength": 1,
          "type": "string",
        },
        "y": {
          "description": "The property name of the Y values for the data of the series.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "x",
        "y",
      ],
      "type": "object",
    },
    "ChartRefreshTarget": {
      "properties": {
        "property": {
          "default": "value",
          "description": "The property in the component state that holds the data for the series.",
          "type": "string",
        },
        "propertyMap": {
          "$ref": "#/definitions/ChartPropertyMap",
        },
        "target": {
          "description": "The property ID of the component that holds the data for the series.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "target",
        "propertyMap",
      ],
      "type": "object",
    },
    "ChartSeries": {
      "properties": {
        "color": {
          "description": "The color of the new series to be displayed in the Chart component. Accepts color hex code, prefixed with #. ex: #990000 for dark red.",
          "examples": [
            "#990000",
          ],
          "type": "string",
        },
        "id": {
          "description": "The unique identifier for the series of the Chart component.",
          "minLength": 1,
          "type": "string",
        },
        "label": {
          "description": "The label of the series. Shown in the Legend and when the user hovers over the chart.",
          "type": "string",
        },
        "refreshTarget": {
          "$ref": "#/definitions/ChartRefreshTarget",
        },
        "stackGroup": {
          "description": "The logical grouping name of the series, which then becomes stacked with other series that use the exact same group name.",
          "type": "string",
        },
        "type": {
          "description": "The type of chart to be displayed for the series.",
          "enum": [
            "area",
            "bar",
            "column",
            "line",
            "pie",
            "scatter",
          ],
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "id",
        "type",
        "refreshTarget",
      ],
      "type": "object",
    },
  },
  "description": "Adds new series to a Chart component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ChartAddSeriesOperationOptions",
    },
    "type": {
      "const": "CHART_ADD_SERIES",
      "examples": [
        "CHART_ADD_SERIES",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
